src/tools/pkcs11-tool.c

```diff
@@ -2081,6 +2081,30 @@ static int unlock_pin(CK_SLOT_ID slot, CK_SESSION_HANDLE sess, int login_type)
 	return 0;
 }
 
+/* return matching ec_curve_info or NULL based on ec_params */
+static const struct ec_curve_info* match_ec_curve_by_params(const unsigned char *ec_params, CK_ULONG ec_params_size)
+{
+	char ecpbuf[64];
+	CK_ULONG ecpsize = ec_params_size*2;
+
+	if (ec_params_size > (sizeof(ecpbuf)/2)) {
+		util_fatal("Invalid EC params");
+	}
+
+	sc_bin_to_hex(ec_params, ec_params_size, ecpbuf, sizeof(ecpbuf), 0);
+
+	for (size_t i = 0; ec_curve_infos[i].name != NULL; ++i) {
+
+		if ((strlen(ec_curve_infos[i].ec_params) == ecpsize) &&
+		    (strcmp(ec_curve_infos[i].ec_params, ecpbuf) == 0))
```

Comment:
I think just `strcmp()` here would be enough, given that both of the strings are correctly null terminated and strcmp checks the length anyway.